automata/pda/pda.py

```diff
@@ -24,6 +52,325 @@ class PDA(Automaton, metaclass=abc.ABCMeta):
+        arrow_size: str,
+    ) -> pgv.AGraph:
+        """
+        Constructs stack for all the transitions in the `input_path` and
+        adds the constructed stacks into `graph`. Returns the same `graph`
+        """
+        from_node = input_path[0][0]
+        label = " | ".join(reversed(from_node.stack))
+        graph.add_node(
+            from_node,
+            label=f" | {label}",
+            shape="record",
+            arrowsize=arrow_size,
+            fontsize=font_size,
+        )
+
+        interpolation = coloraide.Color.interpolate(
+            [start_color, end_color], space="srgb"
+        )
+
+        for i, (c, n) in enumerate(input_path, start=1):
+            from_node = n
+            label = " | ".join(reversed(from_node.stack))
+
+            color = interpolation(i / len(input_path))
+
+            graph.add_node(
+                from_node,
+                label=f" | {label}",
+                shape="record",
+                arrowsize=arrow_size,
+                fontsize=font_size,
+            )
+            graph.add_edge(
+                c,
+                n,
+                label=f"<<b>[<i>#{i}</i>]</b>>",
+                arrowsize=arrow_size,
+                fontsize=font_size,
+                color=color.to_string(hex=True),
+                penwidth="2.5",
+            )
+
+        return graph
+
+    def _create_transitions_for_input_diagram(
+        self,
+        graph: pgv.AGraph,
+        input_path: List[Tuple[PDAConfiguration, PDAConfiguration]],
+        edge_drawn: EdgeDrawnDictT,
+        start_color: coloraide.Color,
+        end_color: coloraide.Color,
+        arrow_size: str,
+        font_size: str,
+    ) -> pgv.AGraph:
+        """
+        Add transitions to show execution of the PDA for the given input string
+        """
+
+        interpolation = coloraide.Color.interpolate(
+            [start_color, end_color], space="srgb"
+        )
+
+        # find all transitions in the finite state machine with traversal.
+        for transition_index, (from_state, to_state) in enumerate(input_path, start=1):
+            color = interpolation(transition_index / len(input_path))
+
+            symbol = self._get_symbol_configuration(from_state, to_state)
+            label = self._get_edge_name(*symbol)
+
+            edge_drawn[from_state.state, to_state.state, symbol] = True
+            graph.add_edge(
+                self._get_state_name(from_state.state),
+                self._get_state_name(to_state.state),
+                label=f"<{label} <b>[<i>#{transition_index}</i>]</b>>",
+                arrowsize=arrow_size,
+                fontsize=font_size,
+                color=color.to_string(hex=True),
+                penwidth="2.5",
+            )
+
+        return graph
+
+    def _add_transitions_diagram(
+        self,
+        graph: pgv.AGraph,
+        is_edge_drawn: EdgeDrawnDictT,
+        arrow_size: str,
+        font_size: str,
+    ) -> pgv.AGraph:
```

Comment:
I think these annotations might cause issues without `from __future__ import annotations` at the top of this file.